packages/zoe/test/unitTests/test-zoe-startInstance.js

```diff
@@ -0,0 +1,158 @@
+  t.deepEqual(Object.getOwnPropertyNames(result).sort(), [
+    'adminFacet',
+    'creatorFacet',
+    'creatorInvitation',
+    'instance',
+    'publicFacet',
+  ]);
+  isEmptyFacet(t, result.creatorFacet);
+  t.deepEqual(result.creatorInvitation, undefined);
+  facetHasMethods(t, result.publicFacet, ['makeInvitation']);
+  t.deepEqual(getStringMethodNames(result.adminFacet), [
+    'getVatShutdownPromise',
+    'restartContract',
+    'upgradeContract',
+  ]);
+});
+
+test('terms, issuerKeywordRecord switched', async t => {
+  const { zoe } = setup();
+  const installation = await E(zoe).installBundleID('b1-contract');
+  const { moolaKit } = setup();
+  await t.throwsAsync(
+    () =>
+      E(zoe).startInstance(
+        installation,
+        { something: 2 },
+        { Moola: moolaKit.issuer },
+      ),
+    {
+      message:
+        'In "startInstance" method of (ZoeService): arg 1?: something: [1]: 2 - Must match one of ["[match:remotable]","[match:kind]"]',
+    },
+  );
+});
+
+test('bad issuer, makeEmptyPurse throws', async t => {
+  const { zoe } = setup();
+  const installation = await E(zoe).installBundleID('b1-contract');
+  const brand = Far('brand', {
+    // eslint-disable-next-line no-use-before-define
+    isMyIssuer: i => i === badIssuer,
+    getDisplayInfo: () => ({ decimalPlaces: 6, assetKind: AssetKind.NAT }),
+  });
+  const badIssuer = Far('issuer', {
+    makeEmptyPurse: async () => {
+      throw Error('bad issuer');
+    },
+    getBrand: () => brand,
+  });
+  await t.throwsAsync(
+    () => E(zoe).startInstance(installation, { Money: badIssuer }),
+    {
+      message:
+        'A purse could not be created for brand "[Alleged: brand]" because: "[Error: bad issuer]"',
+    },
+  );
+});
+
+test('unexpected properties', async t => {
+  const { zoe } = setup();
+
+  const contractPath = `${dirname}/unexpectedPropertiesContract.js`;
+  const bundle = await bundleSource(contractPath);
+  const installation = await E(zoe).install(bundle);
+
+  await t.throwsAsync(() => E(zoe).startInstance(installation), {
+    message:
+      'contract "start" returned unrecognized properties ["unexpectedProperty"]',
+  });
+});
+
+test('prepare and start', async t => {
+  const { zoe } = setup();
+
+  const contractPath = `${dirname}/redundantPrepareContract.js`;
+  const bundle = await bundleSource(contractPath);
+  const installation = await E(zoe).install(bundle);
+
+  await t.throwsAsync(() => E(zoe).startInstance(installation), {
+    message: 'contract must provide exactly one of "start" and "prepare"',
+  });
+});
+
+test('before meta', async t => {
+  const { zoe } = setup();
+
+  const contractPath = `${dirname}/beforeMetaContract.js`;
+  const bundle = await bundleSource(contractPath);
+  const installation = await E(zoe).install(bundle);
+
+  await t.throwsAsync(() => E(zoe).startInstance(installation), {
+    message: 'privateArgs: "[undefined]" - Must be: {"greeting":"hello"}',
+  });
+
```

Comment:
This doesn't seem like a stable test that the contract was parsed correctly. It doesn't seem to be detecting that `prepare` without `meta` is an error.

Reply:
Prepare without meta is correct